### ISSUE TITLE
Loss not going down

### ISSUE THREAD

**Issue by Amir-Arsalan** [wohlert/generative-query-network-pytorch#23]
@wohlert I set `fraction` to `1.0` and trained a model on all training scenes in Shepard Metzler 5 for 110 epochs by setting ADAM's `asmgrad` flag to `True`. What I noticed is that the loss (i.e. ELBO) does not go below 6.97e3 and the reconstructions look very similar (sometimes worse) to the reconstructions obtained from the model you trained on 10% of the data. I also trained the model with your hyperparameters (`asmgrad=False`) on the entire training set of Shepard Metzler 5 and still get very similar reconstructions to what your 10% model is getting. Do you think I might be doing something wrong? The strange thing is that I don't see any issues describing this problem in your repo. I'm assuming that people who have used your code have been able to successfully train a model with more number of epochs on the entire dataset. So I'm not sure why I'm encountering this issue. Just a final note: I am using PyTorch 1.3.1.

Below you can see some reconstructions from your model, the model I trained for 26 epochs on 100% of the data and a model I trained for 110 epochs on 100% of the data:

Your model:
![image](https://user-images.githubusercontent.com/6061721/97312190-99f46580-183b-11eb-8565-454b3d34edfa.png)

trained for 26 epochs:
![image](https://user-images.githubusercontent.com/6061721/97312229-a24ca080-183b-11eb-861a-d6f16b56c896.png)

trained for 110 epochs (looks even worse):
![image](https://user-images.githubusercontent.com/6061721/97312262-aaa4db80-183b-11eb-84e0-1e84fa78539a.png)

Your model:
![image](https://user-images.githubusercontent.com/6061721/97312302-b6909d80-183b-11eb-9d60-17c99c16a6a0.png)

trained for 26 epochs:
![image](https://user-images.githubusercontent.com/6061721/97312316-bbede800-183b-11eb-8d97-558164c1cffd.png)

trained for 110 epochs:
![image](https://user-images.githubusercontent.com/6061721/97312370-cb6d3100-183b-11eb-860c-1c2655853b8d.png)

In the paper however the authors show that the loss should keep going down and the reconstructions should get better over time:
![image](https://user-images.githubusercontent.com/6061721/97310568-c0b19c80-1839-11eb-8847-447c5dc0d405.png)

**Comment by wohlert** [comment]
@Amir-Arsalan https://safenote.co/r/5f98701dbe4bf7@15069468
